frontend/src/main/scala/akka/viz/frontend/components/components.scala

```diff
@@ -183,6 +184,44 @@ class MessageFilter(
 }
 
 class MessagesPanel(selectedActors: Var[Set[String]]) extends Component with PrettyJson {
+  val ShowMoreLength = 200
+
+  private val msgQueue = Var[Queue[Received]](Queue.empty)
+
+  private var lastDisplayed = 0L
+
+  def messageReceived(rcv: Received): Unit = {
+    val selected = selectedActors.now
+    if (selected.contains(rcv.sender) || selected.contains(rcv.receiver)) {
+      if (messagesTbody.childNodes.length < 20) {
+        messagesTbody.appendChild(messageRow(rcv).render)
+        lastDisplayed = rcv.eventId
+      } else {
+        msgQueue() = msgQueue.now.enqueue(rcv)
+      }
+    }
+  }
+
+  val showMoreRow = tr(cell, onclick := displayMoreMessages).render
+  lazy val cell = td(colspan := 3, fontStyle.italic).render
+
+  msgQueue.foreach { q =>
+    if (q.headOption.exists(_.eventId > lastDisplayed))
+      cell.innerHTML = s"${q.length} messages not shown, click to display more"
+  }
+
+
+  def displayMoreMessages: ThisFunction1[TableRow, MouseEvent, Unit] = { (row: TableRow, ev: MouseEvent) =>
+    ev.preventDefault()
+    val (portion, q) = (msgQueue.now.take(ShowMoreLength), msgQueue.now.drop(ShowMoreLength))
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2015979973/issues/source?bid=3121282&fileBranchId=3123640#l216)